openc3/lib/openc3/microservices/text_log_microservice.rb

```diff
@@ -43,22 +43,34 @@ def initialize(name)
       @cycle_time = 600 unless @cycle_time # 10 minutes
       @cycle_size = 50_000_000 unless @cycle_size # ~50 MB
 
+      @tlws = {}
       @error_count = 0
       @metric.set(name: 'text_log_total', value: @count, type: 'counter')
       @metric.set(name: 'text_error_total', value: @error_count, type: 'counter')
     end
 
     def run
       setup_tlws()
+
+      individual_topics = []
+      @topics.each do |topic|
+        individual_topics << [topic]
+      end
+
       while true
         break if @cancel_thread
 
-        Topic.read_topics(@topics) do |topic, msg_id, msg_hash, redis|
+        # Read each topic separately to support multiple redis shards
+        individual_topics.each do |individual_topic|
```

Comment:
Is this an optimization to allow the cancel_thread to break in between topic reads?